src/pages/index.js

```diff
@@ -32,71 +87,145 @@ const cardPopOut = new PopupWithImage({
+  console.log(card);
+  deleteCardModal.open();
+  deleteCardModal.setEventListeners();
+  cardToDelete = card;
+  return cardToDelete;
+  /*
+  return card;
+  cardToDelete = card;
+  console.log(cardToDelete);
+  */
 }
 
-const firstCards = new Section(
-  { items: initialCards, renderer: createCard },
-  ".elements"
-);
+function handleLike() {
+  if (this.data.isLiked == false) {
+    api.addLike(this.data._id);
+    this.data.isLiked = true;
+  } else {
+    api.removeLike(this.data._id);
+    this.data.isLiked = false;
+  }
+}
 
-firstCards.renderItems();
+//Handle edit/add button clicks
 
-const newUserInfo = new UserInfo({
-  userNameSelector: ".info__name",
-  userJobSelector: ".info__job-title",
+modalProfileEditBtn.addEventListener("click", () => {
+  profileModal.open();
+  const data = newUserInfo.getUserInfo();
+  console.log(data);
+  modalName.value = data.name;
+  modalJob.value = data.job;
+  profileValidator.resetValidation();
+});
+
+newProfileImageBtn.addEventListener("click", () => {
+  newProfilePictureModal.open();
+});
+
+modalImageEditBtn.addEventListener("click", () => {
+  newImageModal.open();
 });
 
 const profileModal = new PopupWithForm({
   popupSelector: ".modal_type_profile",
   handleFormSubmit: (formData) => {
-    console.log(formData.name);
-    console.log(formData.job);
-    newUserInfo.setUserInfo({ name: formData.name, job: formData.job });
+    /*
+    profileModal.renderSaving(true);
+    */
+    api
+      .editUserInfo(formData)
+      .then((userData) => {
+        newUserInfo.setUserInfo({ name: userData.name, job: userData.about });
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+    /*
+      .finally(() => profileModal.renderSaving(false));
     profileModal.close();
+    */
   },
+  buttonText: "Save",
+});
+
+const newProfilePictureModal = new PopupWithForm({
+  popupSelector: ".modal_type_change-profile-picture",
+  handleFormSubmit: (formData) => {
+    /*
+    newProfilePictureModal.renderSaving(true);
+    */
+    api
+      .editUserProfilePicture(formData)
+      .then((userData) => {
+        newUserInfo.setUserPicture({ avatar: userData.avatar });
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+    /*
+      .finally(() => newProfilePictureModal.renderSaving(false));
+  
+    newProfilePictureModal.close();
+    */
+  },
+  buttonText: "Save",
 });
 
 const newImageModal = new PopupWithForm({
   popupSelector: ".modal_type_new-image",
   handleFormSubmit: (newData) => {
-    const addedCard = createCard(newData);
-    firstCards.addItem(addedCard);
+    /*
+    newImageModal.renderSaving(true);
+    */
+    api
+      .addNewCard(newData)
+      .then((cardData) => {
+        return createCard(cardData);
+      })
+      .then((addedCard) => cardArea.addItem(addedCard))
+      .catch((err) => {
+        console.error(err);
+      });
     modalImageForm.reset();
     newImageValidator.toggleButtonState();
```

Comment:
They should be in `then`

You need to **reset** the form only  in block `then` in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.


And disabling as well